spiketoolkit/postprocessing/features.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+
+
+def compute_unit_template_features(
+        sorting,
+        recording,
+        unit_ids=None,
+        channel_ids=None,
+        feature_names=None,
+        save_property_or_features=False,
+        recovery_slope_window=0.7,
+        upsampling_factor=1,
+        invert_waveforms=False,
+        as_dataframe=False,
+        verbose=False,
+
+):
+    """
+    Use SpikeInterface/spikefeatures to compute features for the unit
+    template. These consist of a set of 1D features:
+    - peak to valley (peak_to_valley), time between peak and valley
+    - halfwidth (halfwidth), width of peak at half its amplitude
+    - peak trough ratio (peak_trough_ratio), amplitude of peak over amplitude of trough
+    - repolarization slope (repolarization_slope), slope between trough and return to base
+    - recovery slope (recovery_slope), slope after peak towards baseline
+
+    And 2D features:
+
+    The metrics are computed on 'negative' waveforms, if templates are saved as
+    positive, pass keyword 'invert_waveforms'.
+
+    Recording should be bandpassfiltered  # TODO: make this a requirement?
+
+    Parameters:
+    -------
+    sorting: SortingExtractor
+        The sorting result to be evaluated.
+    recording: RecordingExtractor
+        The given recording extractor from which to extract amplitudes
+    unit_ids: list
+        List of unit ids to compute metric for. If not specified, all units are used
+    channel_ids : list
+        if None, max channel for the unit is detected and used. (not used when using template properties)
+    return_dict: bool
+        If True, will return dict of metrics (default False)
+    feature_names: list
+        list of feature names to be computed
+    upsampling_factor : int
+        factor with which to upsample the template resolution (default 1)
+    invert_waveforms : bool
+        invert templates before computing features (default False)
+    recovery_slope_window : float
+        window after peak in ms wherein to compute recovery slope (default 0.7)
+    use_sorting_property_template : bool
+        use templates saved as unit properties (default True)
+
+    save_property_or_features=False,
+    verbose=False,
+
+    Returns
+
+    -------
+    metrics : pandas.DataFrame
+        table containing all metrics:
+            rows: unit_ids
+            columns: metrics
+
+    OR
+
+    metrics : dict
+        dict with metrics as keywords, values are list with
+        metric per unit_id
+    """
+
+    # ------------------- SETUP ------------------------------
+    if unit_ids is None:
+        unit_ids = sorting.get_unit_ids()
+
+    if channel_ids is None:
+        channel_ids = get_unit_max_channels(
+            sorting=sorting,
+            recording=recording,
+            unit_ids=unit_ids,
+        )
+    else:
+        assert len(channel_ids) == len(unit_ids), f'#channel ids should be # unit ids'
```

Comment:
this is wrong. Channel ids independent of unit ids. Imagine you have 100 channels and 10 units: you can select any subset of the 100 channels

Reply:
I'll fix it later :)